### PR TITLE
Make sure to use the correct logger when creating a PackageExtractionContext

### PR DESCRIPTION
**Pull request by PatoBeltran**
## Bug
Fixes: https://github.com/NuGet/Home/issues/7445
Regression: Yes

## Fix
When creating the `PackageExtractionContext` some code paths didn't use the appropriate logger.

This is a regression introduced in 15.9 by https://github.com/NuGet/NuGet.Client/pull/2428. This PR fixes this regression by removing the `PackageExtractionContext` from the `Downloadcontext` and just passing the `ClientPolicyContext`. I also analyzed carefully the changes made in the referenced PR to make sure every other `PackageExtractionContext` has the exact same values for every property as they used to before that change.

### CODE REVIEW

File: src/NuGet.Clients/NuGet.CommandLine/Commands/InstallCommand.cs [hunk 1]
```diff
@@ -388,23 +384,19 @@ private DependencyBehavior GetDependencyBehavior()
                             Console)
                     };
 
+                    if (EffectivePackageSaveMode != Packaging.PackageSaveMode.None)
+                    {
+                        projectContext.PackageExtractionContext.PackageSaveMode = EffectivePackageSaveMode;
```
Comment (jainaashish) [inline src/NuGet.Clients/NuGet.CommandLine/Commands/InstallCommand.cs:389]: how is this changed `projectContext` flowing into `downloadContext` like the previous code?

File: src/NuGet.Core/NuGet.Commands/RestoreCommand/OriginalCaseGlobalPackageFolder.cs [hunk 2]
```diff
@@ -111,7 +110,7 @@ public async Task CopyPackagesToOriginalCaseAsync(IEnumerable<RestoreTargetGraph
                             identity,
                             packageDependency,
                             versionFolderPathResolver,
-                            _request.PackageExtractionContext,
+                            originalCaseContext,
```
Comment (jainaashish) [inline src/NuGet.Core/NuGet.Commands/RestoreCommand/OriginalCaseGlobalPackageFolder.cs:113]: why this change?

File: src/NuGet.Core/NuGet.Commands/RestoreCommand/ProjectRestoreRequest.cs [hunk 3]
```diff
@@ -22,7 +22,14 @@ internal class ProjectRestoreRequest
             ExistingLockFile = existingLockFile;
             MaxDegreeOfConcurrency = request.MaxDegreeOfConcurrency;
             Project = packageSpec;
-            PackageExtractionContext = request.PackageExtractionContext;
+            PackageExtractionContext = new PackageExtractionContext(
+                request.PackageSaveMode,
+                request.XmlDocFileSaveMode,
+                request.ClientPolicyContext,
+                log)
```
Comment (jainaashish) [inline src/NuGet.Core/NuGet.Commands/RestoreCommand/ProjectRestoreRequest.cs:29]: this is so confusing to understand now like where which logger is being used? Could this logger thing be simplified?

File: src/NuGet.Core/NuGet.Commands/RestoreCommand/ProjectRestoreRequest.cs [hunk 4]
```diff
@@ -22,7 +22,14 @@ internal class ProjectRestoreRequest
             ExistingLockFile = existingLockFile;
             MaxDegreeOfConcurrency = request.MaxDegreeOfConcurrency;
             Project = packageSpec;
-            PackageExtractionContext = request.PackageExtractionContext;
+            PackageExtractionContext = new PackageExtractionContext(
```
Comment (jainaashish) [inline src/NuGet.Core/NuGet.Commands/RestoreCommand/ProjectRestoreRequest.cs:25]: If you aren't using `PackageExtractionContext` from `RestoreRequest` then remove that attribute from there since it's a required param to build restore request which is not even going to be used any where

**Review by jainaashish — APPROVED** [review]
I don't know if we can do anything at this time specially for 15.9 but this is so confusing to follow now.

At least add more comments for different loggers at different places, while creating different extractionContexts, for restorecommand, restorerequest, restoerargs...

File: src/NuGet.Clients/NuGet.CommandLine/Commands/InstallCommand.cs (hunk 1 above)
Comment (PatoBeltran) [reply src/NuGet.Clients/NuGet.CommandLine/Commands/InstallCommand.cs:389]: https://github.com/NuGet/NuGet.Client/pull/2428/files#diff-c9a788561af1164b685a274237b25c53L375

File: src/NuGet.Core/NuGet.Commands/RestoreCommand/ProjectRestoreRequest.cs (hunk 4 above)
Comment (PatoBeltran) [reply src/NuGet.Core/NuGet.Commands/RestoreCommand/ProjectRestoreRequest.cs:25]: https://github.com/NuGet/NuGet.Client/pull/2497/files#diff-101420c352b3f5357ea7134fe678d121R25 restore request now doesn't have a `PackageExtractionContext`

File: src/NuGet.Core/NuGet.Commands/RestoreCommand/OriginalCaseGlobalPackageFolder.cs (hunk 2 above)
Comment (PatoBeltran) [reply src/NuGet.Core/NuGet.Commands/RestoreCommand/OriginalCaseGlobalPackageFolder.cs:113]: This is returning to the past logic (https://github.com/NuGet/NuGet.Client/pull/2497/files#diff-101420c352b3f5357ea7134fe678d121R25) because the request doesn't have a `PackageExtractionContext`

File: src/NuGet.Core/NuGet.Commands/RestoreCommand/ProjectRestoreRequest.cs (hunk 3 above)
Comment (PatoBeltran) [reply src/NuGet.Core/NuGet.Commands/RestoreCommand/ProjectRestoreRequest.cs:29]: I don't know enough about this code to propose a way to simplify it. As far as I understand every code path has only one logger. If it doesn't have one, but has access to a project it creates a `LoggerAdapter` to pass as an `ILogger`. The only case that will have more than one is when the `ICollectorLogger` is around, which should be preferred over any other logger.

The cause of this issue was that by creating `PackageExtractionContext` at the beginning and passing it down, we were creating it before the `RestoreCollectorLogger` was created, therefore `PackageExtractionContext` ended up with the wrong logger.

File: src/NuGet.Core/NuGet.Commands/NuGet.Commands.csproj [hunk 5]
```diff
@@ -51,6 +51,18 @@
     </EmbeddedResource>
   </ItemGroup>
 
+  <ItemGroup Condition="$(DefineConstants.Contains(SIGNED_BUILD))">
```
Comment (nkolev92) [inline src/NuGet.Core/NuGet.Commands/NuGet.Commands.csproj:54]: 👏

**Review by nkolev92 — APPROVED** [review]
LGTM
